src/main/java/nz/co/jammehcow/jenkinsdiscord/util/EmbedUtil.java

```diff
@@ -0,0 +1,352 @@
+                EmbedUtil.getFileInputStream(context, step.getCustomFile()),
+                step.getActualDynamicFieldContainer(),
+                step.getEnableArtifactsList(),
+                step.getShowChangeset(),
+                step.getScmWebUrl()
+        );
+    }
+
+    public static WebhookMessage createEmbed(
+            Run<?, ?> build,
+            JenkinsLocationConfiguration globalConfig,
+            TaskListener listener,
+            String title,
+            String link,
+            String description,
+            String footer,
+            String image,
+            String thumbnail,
+            StatusColor statusColor,
+            boolean showStatus,
+            MatrixConfiguration matrixConfiguration,
+            boolean urlLinking,
+            String branch,
+            String notes,
+            String customAvatarUrl,
+            String customUsername,
+            String customFile,
+            InputStream customFileInputStream,
+            DynamicFieldContainer dynamicFieldContainer,
+            boolean enableArtifactsList,
+            boolean showChangeset,
+            String scmWebUrlTemplate
+    ) {
+        String artifactsURL = globalConfig.getUrl() + build.getUrl() + "artifact/";
+
+        WebhookMessageBuilder messageBuilder = new WebhookMessageBuilder();
+        WebhookEmbedBuilder embedBuilder = new WebhookEmbedBuilder();
+
+        messageBuilder.setContent(EmbedUtil.truncateToLimit(listener, "content", notes, EmbedUtil.MAX_CONTENT_LENGTH));
+
+        embedBuilder.setTitle(new EmbedTitle(EmbedUtil.truncateToLimit(listener, "title", title, EmbedUtil.MAX_EMBED_TITLE_LENGTH), link));
+
+        embedBuilder.setDescription(EmbedUtil.truncateToLimit(listener, "description", StringUtils.stripToNull(description), EmbedUtil.MAX_EMBED_DESCRIPTION_LENGTH));
+
+        if (branch != null) {
+            embedBuilder.addField(new EmbedField(true, "Branch", branch));
+        }
+
+        String buildResult = Objects.requireNonNull(build.getResult()).toString().toLowerCase(Locale.ENGLISH);
+        if (urlLinking) {
+            embedBuilder.addField(new EmbedField(true, "Build", MarkdownUtil.formatMarkdownUrl(build.getId(), link)));
+            if (showStatus)
+                embedBuilder.addField(new EmbedField(true, "Status", MarkdownUtil.formatMarkdownUrl(buildResult, link)));
+        } else {
+            embedBuilder.addField(new EmbedField(true, "Build", build.getId()));
+            if (showStatus)
+                embedBuilder.addField(new EmbedField(true, "Status", buildResult));
+        }
+
+        if (matrixConfiguration != null) {
+            StringBuilder descriptionBuilder = new StringBuilder();
+            for (Map.Entry<String, String> entry : matrixConfiguration.getCombination().entrySet())
+                descriptionBuilder.append(String.format("- %s: %s", entry.getKey(), entry.getValue()))
+                                  .append('\n');
```

Comment:
Wouldn't it be simpler to use a `StringJoiner("\n")` here instead of a StringBuilder that will have a trailing newline?

Reply:
I just prefer `StringBuilder` as everyone is able to immediately recognize what it is. Even as someone who is rather experienced with java, I did not initially recognize `StringJoiner` as being a class in the standard library. But if desired, I could replace it with `StringJoiner`. I would however personally prefer `StringBuilder`.